pkg/sigma/v2/pattern.go

```diff
@@ -71,32 +73,56 @@ func NewStringMatcher(
 	lower, all bool,
 	patterns ...string,
 ) (StringMatcher, error) {
-	if patterns == nil || len(patterns) == 0 {
+	if len(patterns) == 0 {
 		return nil, fmt.Errorf("no patterns defined for matcher object")
 	}
 	matcher := make([]StringMatcher, 0)
 	for _, p := range patterns {
-		if strings.HasPrefix(p, "/") && strings.HasSuffix(p, "/") {
-			re, err := regexp.Compile(strings.TrimLeft(strings.TrimRight(p, "/"), "/"))
+		//process modifiers first
+		switch mod {
+		case TextPatternRegex: //regex per spec
+			re, err := regexp.Compile(p)
 			if err != nil {
 				return nil, err
 			}
 			matcher = append(matcher, RegexPattern{Re: re})
-		} else if strings.Contains(p, "*") {
-			matcher = append(matcher, GlobPattern{Token: p, Lowercase: lower})
-		} else {
-			switch mod {
-			case TextPatternRegex: //regex per spec
-				re, err := regexp.Compile(p)
+		case TextPatternContains: //contains: puts * wildcards around the values, such that the value is matched anywhere in the field.
+			p = "*" + p + "*"
+			globNG, err := globng.Compile(p)
+			if err != nil {
+				return nil, err
+			}
+			matcher = append(matcher, GlobPattern{Glob: &globNG})
+		case TextPatternSuffix:
+			matcher = append(matcher, SuffixPattern{Token: p, Lowercase: lower})
+		case TextPatternPrefix:
+			matcher = append(matcher, PrefixPattern{Token: p, Lowercase: lower})
+		default:
+			//no (supported) modifiers, handle non-spec regex, globs and regular values
+			if strings.HasPrefix(p, "/") && strings.HasSuffix(p, "/") {
+				re, err := regexp.Compile(strings.TrimLeft(strings.TrimRight(p, "/"), "/"))
 				if err != nil {
 					return nil, err
 				}
 				matcher = append(matcher, RegexPattern{Re: re})
-			case TextPatternSuffix:
-				matcher = append(matcher, SuffixPattern{Token: p, Lowercase: lower})
-			case TextPatternPrefix:
-				matcher = append(matcher, PrefixPattern{Token: p, Lowercase: lower})
-			default:
+			} else if mod == TextPatternKeyword {
+				//this is a bit hacky, basically if the pattern coming in is a keyword and did not appear
+				//to be a regex, always process it as a 'contains' style glob (can appear anywhere...)
+				//this is due, I believe, on how keywords are generally handled, where it is likely a random
```

Comment:
Yes, exactly. That's the way I see keyword matches as well. For stuff like syslog `MSG` part or event log `Message`, which are basically big text blobs that can contain anything. Sigma keyword I saw as rulewrited saying `I dont care, just match in whatever field is relevant for this sort of stuff` when dealing with structured messages.

So I support the idea of doing general matches over them. Selection type is when we need to be specific.